src/main/java/lotto/domain/GeneratedLotto.java

```diff
@@ -0,0 +1,51 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class GeneratedLotto {
+
+	private static final int LOTTO_NUMBER_MAX_COUNT = 6;
+
+	public List<Integer> lottoNumbers = new ArrayList<>();
+
+	public GeneratedLotto(){
+		Collections.shuffle(Lotto.defaultLottoNumbers);
+		for (int i = 0; i < LOTTO_NUMBER_MAX_COUNT; i++) {
+			lottoNumbers.add(Lotto.defaultLottoNumbers.get(i));
+		}
+	}
+
+	public GeneratedLotto(String[] inputNumber) throws IllegalArgumentException{
+		for (int i = 0; i < inputNumber.length; i++) {
+			Integer number = Integer.parseInt(inputNumber[i].trim());
+			lottoNumbers.add(number);
+		}
+		validateMaxCount(lottoNumbers);
+	}
+
+	public static int countMatchNumber(GeneratedLotto lastWeekPrizeNumberList, GeneratedLotto generatedLotto) {
+		Set<Integer> result = lastWeekPrizeNumberList.lottoNumbers.stream()
+				.distinct()
+				.filter(generatedLotto.lottoNumbers::contains)
+				.collect(Collectors.toSet());
+
+		return result.size();
+	}
+
+	public static void validateMaxCount(List<Integer> generatedLotto) {
```

Comment:
```suggestion
	private void validateMaxCount(List<Integer> generatedLotto) {
```

테스트 코드에서 유효성 검증 메서드를 직접 호출에서 테스트할 필요는 없습니다. 이 메서드를 사용하는 public 메서드에 파라미터로 값을 전달하면서 의도한 예외가 발생하는지 확인하시는걸 추천드립니다